doc/nrf/releases_and_maturity/migration/nRF54H20_migration_2.7/migration_guide_2.4.99-cs3_to_2.7_application.rst

```diff
@@ -0,0 +1,632 @@
+
+   * For :ref:`lib_wifi_credentials` library and Wi-Fi samples:
+
+     * ``CONFIG_WIFI_CREDENTIALS_BACKEND_PSA_UID_OFFSET`` has been removed because it was specific to the previous solution that used PSA Protected Storage instead of PSA Internal Trusted Storage (ITS).
+       Use :kconfig:option:`CONFIG_WIFI_CREDENTIALS_BACKEND_PSA_OFFSET` to specify the key offset for PSA ITS.
+       Be aware that Wi-Fi credentials stored in Protected Storage will not appear in ITS when switching.
+       To avoid re-provisioning Wi-Fi credentials, manually read out the old credentials from Protected Storage in the previously used UID and store to ITS.
+
+zcbor
+=====
+
+.. toggle::
+
+   * If you have zcbor-generated code that relies on the zcbor libraries through Zephyr, you must regenerate the files using zcbor 0.8.1.
+     Note that the names of generated types and members has been overhauled, so the code using the generated code must likely be changed.
+
+     For example:
+
+      * Leading single underscores and all double underscores are largely gone.
+      * Names sometimes gain suffixes like ``_m`` or ``_l`` for disambiguation.
+      * All enum (choice) names have now gained a ``_c`` suffix, so the enum name no longer matches the corresponding member name exactly (because this previously broke the C++ namespace rules).
+
+    * The functions :c:func:`zcbor_new_state`, :c:func:`zcbor_new_decode_state` and the macro :c:macro:`ZCBOR_STATE_D` have gained new parameters related to the decoding of unordered maps.
+      If you are not using this functionality, you can set the functions and the macro to ``NULL`` or ``0``.
+    * The functions :c:func:`zcbor_bstr_put_term` and :c:func:`zcbor_tstr_put_term` have gained a new parameter ``maxlen``, referring to the maximum length of the parameter ``str``.
+      This parameter is passed directly to :c:func:`strnlen` under the hood.
+    * The function :c:func:`zcbor_tag_encode` has been renamed to :c:func:`zcbor_tag_put`.
+    * Printing has been changed significantly, for example, :c:func:`zcbor_print` is now called :c:func:`zcbor_log`, and :c:func:`zcbor_trace` with no parameters is gone, and in its place are :c:func:`zcbor_trace_file` and :c:func:`zcbor_trace`, both of which take a ``state`` parameter.
+
+Recommended changes
+*******************
+
+The following changes are recommended for your application to work optimally after the migration.
+
+General
+=======
+
+.. toggle::
+
+   * Applications that use :file:`prj_<board>.conf` Kconfig configurations should be transitioned to using :file:`boards/<board>.conf` Kconfig fragments.
+
+Application migration examples
+******************************
+
+The following are examples of the changes that were introduced to certain applications to migrate them to the |NCS| v2.7.0.
+
+CoreMark
```

Comment:
Following discussion with @zycz.

Keep in mind that CoreMark sample wasn't present in NCS 2.6.0, so it doesn't have release notes other than "Coremark sample has been added". That's why some crucial info like sample location change has been included here. There's no other place to add this info as we don't have release notes that describe changes between NCS 2.7.0 and 2.4.99-cs3. With that in mind, I wouldn't change anything in this Coremark migration guide. However, if we really want to leave only strictly migration-related points, in next comments I propose changes to be made.